### ISSUE TITLE
Private constructor is accessible outside of the class

### ISSUE THREAD

**Issue by alpercakan** [Microsoft/TypeScript#25546]
**TypeScript Version:**  3.0.0-dev

**Search Terms:** private constructor, constructor access

**Code**

```ts
class E {
  private constructor() { }
}

const x = E.constructor();
```

**Expected behavior:** 
We should get TS2673 `Constructor of class 'E' is private and only accessible within the class declaration.` or TS2341 `Property 'constructor' is private and only accessible within class 'E'.`.

This is important for the singleton pattern. A common way to enforce the singleton pattern is using private constructor, but due to this bug, it is not possible to *completely* restrict instantiation out of the class.

**Actual behavior:** No error is given, the code is compiled to JS successfully.

**Playground Link:** N/A

**Possible Cause:** I have tried some debugging, and it seems that the reason is that the ModifierFlags for the constructor is not kept (even for the private constructor, it seems that `getDeclarationModifierFlagsFromSymbol` gives ModifierFlags.None.

With some guidance, I can also help with fixing the bug.

**Comment by alpercakan** [comment]
@jcalz Oh, I see. I totally agree with you about the confusing naming. Thanks.
